### PR TITLE
Allow guest checkout no matter what

### LINKED ISSUE #10868: Cannot checkout as guest when I have active account
**Sylius version affected**: All

**Description**  
Given I have an enabled customer account, `john@example.com` then I can't checkout with the same email without being logged in.

**Steps to reproduce**  
1. Enable a customer account in the backend
2. Go to checkout and input the same email as the customer you just enabled
3. Try to go to next step.

You will now receive a validation error.

### PR DESCRIPTION
**Pull request by loevgaard**
| Q               | A
| --------------- | -----
| Branch?         | Master
| Bug fix?        | Depends on the eyes ;)
| New feature?    | Again, it depends
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #10868 
| License         | MIT

- [x] Create Behat scenarios
- [x] Implement this change only on checkout (the same behaviour is present in the product review process where we want to keep it)
- [x] Update the docs describing this behaviour. Remember, also, to include a guide to add the 'registered user validation' again

### CODE REVIEW

File: docs/book/orders/checkout.rst [hunk 1]
```diff
@@ -76,10 +76,26 @@ How to perform the Addressing Step programmatically?
...
 
 * An already logged in **User** - the Customer is set for the Order using the `CartBlamerListener <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/EventListener/CartBlamerListener.php>`_, that determines the user basing on the event.
-* An existent **User** that is not logged in - If there is an account in the system registered under the e-mail that has been provided - they are asked for a password to log in before continuing inside the addressing form.
-* A **Customer** that was present in the system before (we've got their e-mail) - the Customer instance is updated via cascade, the order is assigned to it.
+* A **Customer** or **User** that was present in the system before (we've got their e-mail) - the Customer instance is updated via cascade, the order is assigned to it.
 * A new **Customer** with unknown e-mail - a new Customer instance is created and assigned to the order.
 
+.. note::
+
+    Before Sylius ``v1.7`` a **User** (i.e. we have their e-mail and they are registered) had to login before they could complete the checkout process. If you want this constraint on your checkout, you can add this to your application:
```
Comment (Zales0123) [inline docs/book/orders/checkout.rst]: I wonder if we want to have a configuration checkbox on channel level about it? Similar to `Is account verification required?` cc @CoderMaggie

Comment (loevgaard) [reply docs/book/orders/checkout.rst]: I chatted with Maggie about that, and she said no, but it could be an option to have it.

**Comment by pamil** [comment]
Thank you, Joachim! :1st_place_medal:

**Comment by loevgaard** [comment]
Great! Thank you very much guys!

**Pull request merged**